RSSReader/Presenter/FeedPresenter.m

```diff
@@ -0,0 +1,66 @@
+//
+//  FeedPresenter.m
+//  RSSReader
+//
+//  Created by Владислав Станкевич on 19.11.20.
+//
+
+#import "FeedPresenter.h"
+#import "NetworkManager.h"
+#import "RSSParser.h"
+
+@interface FeedPresenter ()
+
+@property (nonatomic, copy) void (^completion)(BOOL);
+@property (nonatomic, retain) NetworkManager *networkManager;
+@property (nonatomic, retain) RSSParser *parser;
+@property (nonatomic, retain) NSMutableArray* feedItemArray;
+
+@end
+
+@implementation FeedPresenter
+
+- (id)initWithArray:(NSMutableArray *)array networkManager:(NetworkManager *)manager parser:(RSSParser *)parser{
+    if (self = [super init]) {
+        if (array) {
+            _feedItemArray = array;
+        }
+        else {
+            _feedItemArray = [NSMutableArray new];
+        }
+        _networkManager = manager;
+        _parser = parser;
+    }
+    return self;
+}
+
+- (void)loadNewsWithCompletion:(void (^)(NSError *))completion {
+    [self.networkManager loadFeedWithCompliteon:^(NSData * data, NSError *error) {
+        if (error) {
+            completion(error);
+            NSLog(@"%@", error);
+        }
+        else {
+            [self parseRowData:data completion:^(NSError *error) {
```

Comment:
It would be better to move parsing inside a closure passed as an input parameter from outside.

Reply:
I understand, but idk how to implement this changes correctly... so I changed this method just a little and I hope that this is not critical